lib/worker/src/routes/vaults.rs

```diff
@@ -0,0 +1,460 @@
+            }
+
+            // if we could not parse the datetime, use the default
+            return 0;
+        }
+        0
+    }
+
+    fn after(&self) -> i64 {
+        if let Some(after) = &self.after {
+            let after = match after.parse::<i64>() {
+                Ok(timestamp) => return timestamp,
+                Err(_) => after,
+            };
+
+            if let Ok(after) = DateTime::parse_from_str(after.as_str(), "%Y-%m-%d") {
+                return after.timestamp();
+            }
+
+            if let Ok(after) = DateTime::parse_from_rfc3339(after.as_str()) {
+                return after.timestamp();
+            }
+
+            // if we could not parse the datetime, use the default
+            return 0;
+        }
+        0
+    }
+}
+
+pub async fn create_vault<E: EVMClient + 'static + std::marker::Sync>(
+    path: web::Path<String>,
+    input: web::Form<CreateVaultInput>,
+    evm_client: web::Data<E>,
+    pool: web::Data<PgPool>,
+) -> HttpResponse {
+    let vault: Vault = match path.try_into() {
+        Ok(p) => p,
+        Err(err) => {
+            return HttpResponse::BadRequest().json(ErrorResponse {
+                error: err.to_string(),
+            })
+        }
+    };
+
+    let account: String = match &input.account {
+        Some(v) => v.to_string(),
+        None => {
+            return HttpResponse::BadRequest().json(ErrorResponse {
+                error: "account is empty".to_string(),
+            })
+        }
+    };
+
+    let address: Address = match Address::from_str(&account) {
+        Ok(v) => v,
+        Err(err) => {
+            log::error!("{}", err);
+            return HttpResponse::BadRequest().json(ErrorResponse {
+                error: "not a valid account".to_string(),
+            });
+        }
+    };
+
+    match evm_client
+        .add_pub(address, vault.to_string().as_str())
+        .await
+    {
+        Ok(_) => {}
+        Err(err) => {
+            log::error!("{}", err);
+            return HttpResponse::BadRequest().json(ErrorResponse {
+                error: "failed to create vault".to_string(),
+            });
+        }
+    }
+
+    let created = match db::namespace_create(
+        &pool,
+        &vault.namespace(),
+        &vault.relation(),
+        input.cache,
+        address,
+    )
+    .await
+    {
+        Ok(v) => v,
+        Err(err) => {
+            log::error!("{}", err);
+            return HttpResponse::BadRequest().json(ErrorResponse {
+                error: "failed to create vault".to_string(),
+            });
+        }
+    };
+
+    HttpResponse::Ok().json(CreateVaultResponse { created })
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+pub struct CreateVaultInput {
+    pub cache: Option<i64>,
+    pub account: Option<String>,
+}
+
+#[derive(Debug, Serialize)]
+pub struct CreateVaultResponse {
+    created: bool,
+}
+
+pub async fn write_record(
+    path: web::Path<String>,
+    params: web::Query<WriteRecordParams>,
```

Comment:
this might work if you want to pass `WriteRecordParams` in form data: https://actix.rs/docs/extractors#url-encoded-forms